cluster.go

```diff
@@ -166,7 +171,7 @@ func newClusterNode(clOpt *ClusterOptions, addr string) *clusterNode {
 	opt := clOpt.clientOptions()
 	opt.Addr = addr
 	node := clusterNode{
-		Client: NewClient(opt),
+		Client: clOpt.NewClient(opt),
```

Comment:
what would be nice would be to have the limiter take the address as a parameter. If one shard is down then we can have a circuit breaker that disables that shard only